### ISSUE TITLE
Update Google Guava to 31.1-jre

### ISSUE THREAD

**Issue by selundqma** [google/guice#1642]
Update dependency Google Guava to [31.1](https://github.com/google/guava/tree/v31.1)

**Comment by selundqma** [comment]
Dear guice developers. When do you think this update can be done and delivered?

**Comment by HannesWell** [comment]
In https://github.com/google/guice/commit/415a377e3bb7fe4497c3ee9f573d2652c1651038 guava was updated to guava 31.0.1. That's not exactly 31.1 but at least its in the 31 line.

**Comment by cdietrich** [comment]
is there a plan for a guava 32 based guice version?

**Comment by sameb** [comment]
No, there is no plan for a new release with Guava 32. Guice does not shade
the guava dependency, so it should be trivial to exclude the transitive
Guava dependency and instead explicitly include the desired version of
Guava. (Binary incompatible changes are extremely rare in Guava --
particularly in the small subset of Guava that Guice uses -- and
re-releasing Guice whenever Guava releases isn't sustainable.)

sam
